mobile-app/app/screens/AppNavigator/screens/Loans/screens/ConfirmEditCollateralScreen.tsx

```diff
@@ -6,49 +6,87 @@ import BigNumber from 'bignumber.js'
 import { hasTxQueued as hasBroadcastQueued } from '@store/ocean'
+import { TokenData } from '@defichain/whale-api-client/dist/api/tokens'
+import { NativeLoggingProps, useLogger } from '@shared-contexts/NativeLoggingProvider'
+import { WhaleWalletAccount } from '@defichain/whale-api-wallet'
+import { CTransactionSegWit, TransactionSegWit } from '@defichain/jellyfish-transaction'
+import { CollateralItem } from '@screens/AppNavigator/screens/Loans/screens/EditCollateralScreen'
+import { useCollateralPrice } from '@screens/AppNavigator/screens/Loans/hooks/CollateralPrice'
+import { onTransactionBroadcast } from '@api/transaction/transaction_commands'
+import { fetchVaults } from '@store/loans'
+import { useWalletContext } from '@shared-contexts/WalletContext'
+import { useWhaleApiClient } from '@shared-contexts/WhaleContext'
+import { ConversionTag } from '@components/ConversionTag'
+import { ConversionParam } from '@screens/AppNavigator/screens/Balances/BalancesNavigator'
 
 type Props = StackScreenProps<LoanParamList, 'ConfirmEditCollateralScreen'>
 
-export function ConfirmEditCollateralScreen ({ route, navigation }: Props): JSX.Element {
+export function ConfirmEditCollateralScreen ({
+  route,
+  navigation
+}: Props): JSX.Element {
   const {
-    vaultId,
-    collaterals,
-    totalCollateralValue,
-    fee
+    vault,
+    token,
+    amount,
+    fee,
+    isAdd,
+    collateralItem,
+    conversion
   } = route.params
+  const { address } = useWalletContext()
+  const client = useWhaleApiClient()
   const hasPendingJob = useSelector((state: RootState) => hasTxQueued(state.transactionQueue))
   const hasPendingBroadcastJob = useSelector((state: RootState) => hasBroadcastQueued(state.ocean))
+  const [isOnPage, setIsOnPage] = useState<boolean>(true)
+  const dispatch = useDispatch()
+  const logger = useLogger()
+
+  useEffect(() => {
+    setIsOnPage(true)
+    return () => {
+      setIsOnPage(false)
+    }
+  }, [])
 
   function onCancel (): void {
     navigation.navigate({
       name: 'EditCollateralScreen',
       params: {
-        vaultId
+        vaultId: vault.vaultId
       },
       merge: true
     })
   }
 
   async function onSubmit (): Promise<void> {
-    // TODO: create signer to add collateral, remove custom navigation below
-    navigation.navigate({
-      name: 'VaultDetailScreen',
-      params: {
-        vaultId: vaultId,
-        emptyActiveLoans: false
-      },
-      merge: true
-    })
+    if (hasPendingJob || hasPendingBroadcastJob) {
+      return
+    }
+    if (isAdd) {
+      await addCollateral({
+        vaultId: vault.vaultId,
+        token,
+        tokenAmount: amount
+      }, dispatch, logger, () => {
+        onTransactionBroadcast(isOnPage, navigation.dispatch, 1)
+      }, () => {
+        dispatch(fetchVaults({
+          address,
+          client
+        }))
+      })
+    }
   }
 
   function getSubmitLabel (): string {
```

Comment:
need to handle the condition to return `CONVERTING` label

Reply:
will update